TestEnvironment/demo-static-parent-issue/MyClass.php

```diff
@@ -0,0 +1,22 @@
+<?php
+class MyClass extends ParentClass
+{
+
+	public function load()
+	{
+		$dataProvider = DataProviderWithStaticFactory::getDataProvider();
+		$entity = $dataProvider->returnData();
+
+		$refreshedEntity = self::refreshEntity($entity);
+		$refreshedEntity->entityMethod(); // auto-complete does not work for this
```

Comment:
Try the upload now.

The one I did before was limited to 1 level on method calls. I enabled
multiple levels of method call . The parameter internally is really read as
DataProviderWithStaticFactory::getDataProvider()->returnData() . Have fun
anyway.




.


On 26 April 2016 at 15:22, Martin Hujer <notifications@github.com> wrote:

> In TestEnvironment/demo-static-parent-issue/MyClass.php
> <https://github.com/pbyrne84/DynamicReturnTypePluginTestEnvironment/pull/3#discussion_r61094013>
> :
>
> > @@ -0,0 +1,22 @@
> > +<?php
> > +class MyClass extends ParentClass
> > +{
> > +
> > +	public function load()
> > +	{
> > +		$dataProvider = DataProviderWithStaticFactory::getDataProvider();
> > +		$entity = $dataProvider->returnData();
> > +
> > +		$refreshedEntity = self::refreshEntity($entity);
> > +		$refreshedEntity->entityMethod(); // auto-complete does not work for this
>
> @pbyrne84 <https://github.com/pbyrne84> just to be sure - this PR is not
> another repro-case for previous issue, but rather an example of a different
> (but related) issue with static / parent calls
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pbyrne84/DynamicReturnTypePluginTestEnvironment/pull/3/files/efd40e1691b3bebe9b67e300f84147ef8feab975#r61094013>
>